src/test/java/uk/gov/justice/hmpps/prison/aop/connectionproxy/OracleConnectionAspectTest.java

```diff
@@ -33,56 +40,105 @@ class OracleConnectionAspectTest {
 
-        connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection);
+                ArgumentCaptor<Properties> propertiesCaptor = ArgumentCaptor.forClass(Properties.class);
+                verify(proxyConnection).openProxySession(eq(OracleConnection.PROXYTYPE_USER_NAME), propertiesCaptor.capture());
+                assertThat(propertiesCaptor.getValue().get((OracleConnection.PROXY_USER_NAME))).isEqualTo("some user name");
+            }
 
-        verify(proxyConnection, never()).openProxySession(eq(OracleConnection.PROXYTYPE_USER_NAME), any(Properties.class));
-    }
+            @Test
+            void setSchemaAndContext() throws SQLException {
+                configureMocks(AuthSource.NOMIS, "");
+
+                connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection);
+
+                ArgumentCaptor<String> sqlCaptor = ArgumentCaptor.forClass(String.class);
+                verify(pooledConnection, times(2)).prepareStatement(sqlCaptor.capture());
+                assertThat(sqlCaptor.getAllValues().get(0)).contains("ALTER SESSION SET CURRENT_SCHEMA");
+                assertThat(sqlCaptor.getAllValues().get(1)).contains("nomis_context.set_client_nomis_context");
+            }
+
+        }
+
+        @Nested
+        class ProxyUser {
+            @Test
+            void doesntOpenProxyConnection() throws SQLException {
+                configureMocks(AuthSource.NONE, "some_user");
+
+                connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection);
+
+                verify(proxyConnection, never()).openProxySession(eq(OracleConnection.PROXYTYPE_USER_NAME), any(Properties.class));
+            }
 
-    @Test
-    void openProxySessionIfIdentifiedAuthentication_notANomisUser_setSchemaButNotContext() throws SQLException {
-        configureMocks(AuthSource.NONE);
+            @Test
+            void setSchemaAndContext() throws SQLException {
+                configureMocks(AuthSource.NONE, "some_user");
 
-        connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection);
+                connectionAspect.openProxySessionIfIdentifiedAuthentication(pooledConnection);
 
-        ArgumentCaptor<String> sqlCaptor = ArgumentCaptor.forClass(String.class);
-        verify(pooledConnection, times(1)).prepareStatement(sqlCaptor.capture());
-        assertThat(sqlCaptor.getAllValues().get(0)).contains("ALTER SESSION SET CURRENT_SCHEMA");
+                ArgumentCaptor<String> sqlCaptor = ArgumentCaptor.forClass(String.class);
+                verify(pooledConnection, times(2)).prepareStatement(sqlCaptor.capture());
+                assertThat(sqlCaptor.getAllValues().get(0)).contains("nomis_context.set_client_nomis_context");
+                assertThat(sqlCaptor.getAllValues().get(1)).contains("ALTER SESSION SET CURRENT_SCHEMA");
+            }
+        }
+
+        @Nested
+        class NotNomisOrProxyUser {
```

Comment:
GitHub struggled with the diff for the unit tests 😞 But essentially the new tests are in the class.